lib/components/src/tooltip/TooltipLinkList.tsx

```diff
@@ -50,16 +53,88 @@ const Item: FunctionComponent<TooltipLinkListProps['links'][number]> = (props) =
   );
 };
 
+function nextItem(list: HTMLUListElement, item: HTMLElement): HTMLElement {
+  if (list === item) {
+    return list.firstChild as HTMLElement;
+  }
+  if (item && item.nextElementSibling) {
+    return item.nextElementSibling as HTMLElement;
+  }
+  return list.firstChild as HTMLElement;
+}
+
+function previousItem(list: HTMLUListElement, item: HTMLElement): HTMLElement {
+  if (list === item) {
+    return list.lastChild as HTMLElement;
+  }
+  if (item && item.previousElementSibling) {
+    return item.previousElementSibling as HTMLElement;
+  }
+  return list.lastChild as HTMLElement;
+}
+
+function moveFocus(
+  list: HTMLUListElement,
+  currentFocus: HTMLElement,
+  traversalFunction: (list: HTMLUListElement, item: Element) => HTMLElement | null
+) {
+  let wrappedOnce = false;
+  const nextFocus = traversalFunction(list, currentFocus);
+
+  while (nextFocus) {
+    // Prevent infinite loop.
+    if (nextFocus === list.firstChild) {
+      if (wrappedOnce) {
+        return;
+      }
+      wrappedOnce = true;
+    }
+
+    nextFocus.focus();
+    return;
+  }
+}
+
 export const TooltipLinkList: FunctionComponent<TooltipLinkListProps> = ({
   links,
   LinkWrapper,
-}) => (
-  <List>
-    {links.map(({ isGatsby, ...p }) => (
-      <Item key={p.id} LinkWrapper={isGatsby ? LinkWrapper : null} {...p} />
-    ))}
-  </List>
-);
+}) => {
+  const listRef = React.useRef<HTMLUListElement>(null);
+
+  const handleKeyDown: React.KeyboardEventHandler<HTMLUListElement> = (event) => {
+    const list = listRef.current;
+    const { key } = event;
+    const currentFocus = ((list && list.ownerDocument) || document).activeElement;
```

Comment:
You'll have to test that there's no modifier keys active. Otherwise they'll conflict with other key listeners. Example:

https://github.com/storybookjs/storybook/blob/adae4e5011d756927d60ff8139cb01e7b07cb89e/lib/ui/src/components/sidebar/useHighlighted.ts#L83

Reply:
I don't think this is right / needed, I've tested with both and didn't changed anything for me.
Keyboard navigation is working in menus.
Keyboard navigation is working between stories with / without a menu open